swap_meet/vendor.py

```diff
@@ -0,0 +1,90 @@
+from .item import Item
+import operator
+
+class Vendor:
+
+
+    def __init__(self, inventory = None):
+        '''constructs instance of Vendor, 
+        initializing inventory with default None'''
+        
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        '''receives item as an argument and adds it to Vendor inventory'''
+        
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        '''receives item as an argument and removes it from Vendor inventory'''
+        
+        if item in self.inventory:
+            self.inventory.remove(item)
+        else:
+            return False
+        return item
+    
+    def get_by_category(self, category):
+        '''receives category as an argument,
+        creates and returns a list of items in that category'''
+
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+    
+    def swap_items(self, other_vendor, item_a, item_b):
+        '''receives other vendor and items as arguments,
+        exchanges those items using add and remove methods'''
+
+        if item_a in self.inventory and item_b in other_vendor.inventory:
+            other_vendor.add(item_a)
+            self.remove(item_a)
+            other_vendor.remove(item_b)
+            self.add(item_b)
+            return True
+        else:
+            return False
```

Comment:
In another situation, you might have many many folks at this swap meet, and item_b is a really desired thing. While you are swapping it with someone, what's to prevent someone else from swooping in and taking it before you?

In that situation, we'd need something that would ensure only you and the other person involved in the swap basically had 'dibs'. This would be a case to use something called a 'lock'. A lock effectively makes it so only you have the ability to change that variable.

This is an advanced topic but I thought I'd mention it here so when you hear about it later it might ring a bell.